### PR TITLE
(DOCSP-27160): Flutter Test and Debug page

### PR DESCRIPTION
**Pull request by mongodben**
## Pull Request Info

Flutter Test and Debug page

### Jira

- https://jira.mongodb.org/browse/DOCSP-27160

### Staged Changes

- [Test & Debug (Flutter)](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-27160/sdk/flutter/test-and-debug/)

### Reminder Checklist

If your PR modifies the docs, you might need to also update some corresponding
pages. Check if completed or N/A.

- [x] Create Jira ticket for corresponding docs-app-services update(s), if any
- [x] Checked/updated Admin API
- [x] Checked/updated CLI reference

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/sdk/flutter/test-and-debug.txt [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+----
+
+To run tests on the Flutter SDK using the
+`flutter_test <https://api.flutter.dev/flutter/flutter_test/flutter_test-library.html>`__
+and `test <https://dart.dev/tools/dart-test>`__, you **must** first run
+the following command:
+
+.. code:: shell
+
+   dart run realm install
```
Comment (desistefanova) [inline source/sdk/flutter/test-and-debug.txt:29]:
```suggestion
   flutter pub run realm install
```

**Review by dacharyc — APPROVED** [review]
Awesome work - I didn't spot any blocking changes. 

I did have a couple of notes about the Test Synced Realms section, but that could be implemented in a future unit of work.

One thing I don't see here and I wonder if we want to add it - why use an actual realm instead of mocking? Particularly for Sync/App Services. Maybe it's reasonable to assume that if somebody wants this page, they're looking for info about integration tests and there's no point in mentioning mocking. Just wanted to throw it out there.

File: source/sdk/flutter/test-and-debug.txt [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+Test Synced Realms
+~~~~~~~~~~~~~~~~~~
+
+To test how your Flutter app works with :ref:`Atlas Device Sync <flutter-sync>`,
+create a separate App Services App with the same configuration as your production app,
+and connect to it from your test file. To learn more about how to make a copy of the App,
+refer to :ref:`Copy an App <copy-realm-app>` in the App Services documentation.
+
+Testing synced realms builds on the Clean up Realm Database in Tests and
```
[the remark this answers is not in the document]
Comment (mongodben) [reply source/sdk/flutter/test-and-debug.txt:121]: thanks, great points! 

per your suggestions, i'm going to:

1. add Realm.deleteAll() to the sync example
2. use Realm.deleteUser() in the app services examples

**Comment by mongodben** [comment]
> One thing I don't see here and I wonder if we want to add it - why use an actual realm instead of mocking? Particularly for Sync/App Services. Maybe it's reasonable to assume that if somebody wants this page, they're looking for info about integration tests and there's no point in mentioning mocking. Just wanted to throw it out there.

my thought on this is that it'd be pretty hard to faithfully mock realm, especially w device sync. do you have any thoughts on what advice on mocking realm/device sync would look like?

**Comment by dacharyc** [comment]
> > One thing I don't see here and I wonder if we want to add it - why use an actual realm instead of mocking? Particularly for Sync/App Services. Maybe it's reasonable to assume that if somebody wants this page, they're looking for info about integration tests and there's no point in mentioning mocking. Just wanted to throw it out there.
> 
> my thought on this is that it'd be pretty hard to faithfully mock realm, especially w device sync. do you have any thoughts on what advice on mocking realm/device sync would look like?

Yeah, I don't disagree. One thought is just to have a section that says that - i.e. "mocking realm and/or Device Sync faithfully is really difficult - prefer integration testing for ease."

Nate and I talked about this back when he was working on a testing page for Java. There may be stuff to borrow from it? [Testing - Java SDK](https://www.mongodb.com/docs/realm/sdk/java/advanced-guides/testing/). It _looks_ like he actually set up a mocking library to mock Realm DB in unit tests. It would probably be good to explore what this would look like for _all_ the SDKs because that seems pretty valuable to me.

**Comment by mongodben** [comment]
> Nate and I talked about this back when he was working on a testing page for Java. There may be stuff to borrow from it? [Testing - Java SDK](https://www.mongodb.com/docs/realm/sdk/java/advanced-guides/testing/). It _looks_ like he actually set up a mocking library to mock Realm DB in unit tests. It would probably be good to explore what this would look like for _all_ the SDKs because that seems pretty valuable to me.

oh that's cool. thx for bringing that up. i'd never seen smthn like this where you pass the class, and the mock can just mock it. want to see if smthn like that exists for Dart/Flutter ecosystem.

**Pull request merged**